code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -295,11 +299,15 @@
 					if(actual)
 						if(!cell.use(actual / powerefficiency))
 							say("Not enough energy to complete operation!")
-							return
+							earlyabort = TRUE
+							break
 						R.add_reagent(reagent, actual)
 						work_animation()
 				else
 					recording_recipe[key] += dispense_amount
+			logstring = logstring.Join(", ")
+			if(!recording_recipe)
+				log_reagent("DISPENSER: [key_name(usr)] dispensed recipe [params["recipe"]] with chemicals [logstring] to [beaker] ([REF(beaker)])[earlyabort? " (aborted early)":""]")
```

Comment:
Earlyabort is always true when reaching this, that indended?

Reply:
OOF no